src/app/services/push.service.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
+
+import { ApiResponse } from '../models/api-response';
+import { getStatuses, getMetadata } from '../utils';
+@Injectable({
+	providedIn: 'root'
+})
+export class PushService {
+
+	constructor( private http: HttpClient ) { }
+
+	public postData(data: ApiResponse) {
+		const jsonpUrl = 'https://api.loklak.org/api/push.json';
+		const headers = new HttpHeaders({
+			'Content-Type': 'application/x-www-form-urlencoded',
+			'Accept': 'application/json',
+			'cache-control': 'no-cache'
+			});
+		if (data) {
+				const temp = '' + JSON.stringify(data);
+				if (JSON.parse(temp)['aggregations']) {
+					const subTemp = temp.substr(0, temp.length - 19) + '}';
+					if (JSON.parse(subTemp)['statuses'][0]) {
+						const metadata = '{"search_metadata":' + getMetadata(subTemp) + ',"statuses":';
```

Comment:
Not sure what this is about but this is not the right way to generate JSON strings.

Something as simple as this should do - 

```js
const {metadata, statuses} = data;
const dataToSend = JSON.stringify({search_metadata: metadata, statuses});
```